lib/IRGen/GenProto.cpp

```diff
@@ -3618,6 +3618,24 @@ void irgen::bindFromGenericRequirementsBuffer(IRGenFunction &IGF,
   }
 }
 
+static llvm::Type *
+getTypeForGenericRequirement(IRGenModule &IGM, GenericRequirement requirement) {
+  switch (requirement.getKind()) {
+  case GenericRequirement::Kind::Shape:
+    return IGM.SizeTy;
+  case GenericRequirement::Kind::Metadata: {
+    CanGenericTypeParamType ty;
+    if ((ty = dyn_cast<GenericTypeParamType>(requirement.getTypeParameter())) &&
+        ty->isParameterPack())
```

Comment:
We might see this kind of requirement for associated types of packs, too, right?

You should have enough information to set the requirement kind as `MetadataPack` / `WitnessTablePack` in the first place, and that'll probably make clients of this a lot easier.

Reply:
Added these new kinds in https://github.com/apple/swift/pull/63363/commits/6a44de214ddaafb216c8fd85acb9bb79bd6ee296 .  Does the static member function `GenericRequirement::isPack` added there correctly identify which types are packs?